src/ducks/loans/allLiquidations.js

```diff
@@ -0,0 +1,175 @@
+			state.loadingError = action.payload.error;
+			state.isLoading = false;
+			state.isRefreshing = false;
+		},
+		fetchLiquidationsSuccess: (state, action) => {
+			state.liquidations = action.payload.liquidations;
+			state.isLoading = false;
+			state.isRefreshing = false;
+			state.isLoaded = true;
+		},
+	},
+});
+
+export const getLiquidationsState = (state) => state.loans.allLiquidations;
+export const getLiquidationsMap = (state) => getLiquidationsState(state).liquidations;
+export const getIsLoadingLiquidations = (state) => getLiquidationsState(state).isLoading;
+export const getIsRefreshingLiquidations = (state) => getLiquidationsState(state).isRefreshing;
+export const getIsLoadedLiquidations = (state) => getLiquidationsState(state).isLoaded;
+export const getLiquidationsLoadingError = (state) => getLiquidationsState(state).loadingError;
+
+export const getLiquidations = createSelector(getLiquidationsMap, (liquidationsMap) =>
+	Object.values(liquidationsMap)
+);
+
+const {
+	fetchLiquidationsRequest,
+	fetchLiquidationsSuccess,
+	fetchLiquidationsFailure,
+} = liquidationsSlice.actions;
+
+const fetchPartialLiquidations = async (loanId) => {
+	const partialLiquidations = await pageResults({
+		api: loansGraph,
+		query: {
+			entity: 'loanPartiallyLiquidateds',
+			selection: {
+				where: {
+					loanId: `\\"${loanId}\\"`,
+				},
+			},
+			properties: [
+				'account',
+				'liquidatedAmount',
+				'liquidator',
+				'liquidatedCollateral',
+				'timeStamp',
+				'loanId',
+				'id',
+			],
+		},
+	});
+
+	const mappedPartialLiquidations = partialLiquidations.map((partial) => {
+		const liquidatedCollateral = bigNumberFormatter(partial.liquidatedCollateral);
+		const penaltyAmount = liquidatedCollateral * 0.1;
+		return {
+			txHash: partial.id,
+			liquidator: partial.liquidator,
+			liquidatedAmount: bigNumberFormatter(partial.liquidatedAmount),
+			liquidatedCollateral: liquidatedCollateral,
+			penaltyAmount: penaltyAmount,
+			loanId: Number(partial.loanID),
+		};
+	});
+
+	return mappedPartialLiquidations;
+};
+
+export const fetchLiquidations = () => async (dispatch, getState) => {
+	const {
+		snxJS: { EtherCollateralsUSD },
+	} = snxJSConnector;
+
+	const state = getState();
+	const ethRate = getEthRate(state);
+
+	let contract = EtherCollateralsUSD.contract;
+	const C_RATIO = bigNumberFormatter(await contract.collateralizationRatio()) / 100;
+	const PENALTY = bigNumberFormatter(await contract.liquidationPenalty());
+
+	dispatch(fetchLiquidationsRequest());
+
+	try {
+		let loans = await pageResults({
+			api: loansGraph,
+			query: {
+				entity: 'loans',
+				selection: {
+					where: {
+						isOpen: true,
+						collateralMinted: '\\"sUSD\\"',
+					},
+				},
+				properties: ['account', 'amount', 'id', 'isOpen', 'hasPartialLiquidations'],
+			},
+		});
+
+		const liquidatedLoans = loans.map(async (loan) => {
+			const id = Number(loan.id);
+			const loanMetaData = await contract.getLoan(loan.account, id);
```

Comment:
@clementbalestrat would snx-data be able to access the contract to get the loan? 

Currently it needs to get the graph data -> but the data that graph returns does not include the current collateral amount. So that's why there's an additional contract.method() call to get it.